Examples/OISTVisualizationTool/main.swift

```diff
@@ -75,6 +80,38 @@ func makeOISTBatch(dataset: OISTBeeVideo, appearanceModelSize: (Int, Int), batch
   return (statistics.normalized(stacked), statistics)
 }
 
+/// Returns a `[[h, w, c]]` batch of normalized patches from a VOT video, and returns the
+/// statistics used to normalize them.
+/// - dataset: Bee video dataset object
+/// - appearanceModelSize: [H, W]
+/// - batchSize: number of batch samples
+/// - seed: Allow controlling the random sequence
+/// - trainSplit: Controls where in the frames to split between train and test
+func makeOISTTrainingBatch(dataset: OISTBeeVideo, appearanceModelSize: (Int, Int), batchSize: Int = 300, seed: Int = 42, trainSplit: Int = 250)
+  -> (normalized: [Tensor<Double>], statistics: FrameStatistics)
+{
+  var images: [Tensor<Double>] = []
+  images.reserveCapacity(batchSize)
+
+  var currentFrame: Tensor<Double> = [0]
+  var currentId: Int = -1
+
+  var statistics = FrameStatistics(Tensor<Double>([0.0]))
+  statistics.mean = Tensor(62.26806976644069)
+  statistics.standardDeviation = Tensor(37.44683834503672)
+
+  var deterministicEntropy = ARC4RandomNumberGenerator(seed: seed)
+  for label in dataset.labels[0..<trainSplit].randomSelectionWithoutReplacement(k: 10, using: &deterministicEntropy).lazy.joined().randomSelectionWithoutReplacement(k: batchSize, using: &deterministicEntropy).sorted(by: { $0.frameIndex < $1.frameIndex }) {
```

Comment:
Does the Batches data structure that I pointed to in the notebook not do all this sampling already? Seems like reinventing the wheel here, but that's only from the cursory glance I can afford at the moment.